framework/scene_graph/components/image.cpp

```diff
@@ -312,10 +319,41 @@ std::vector<uint8_t> &Image::get_mut_data()
 	return data;
 }
 
+void Image::update_hash()
+{
+	static_assert(sizeof(data[0]) == 1);
+	constexpr size_t chunk_size = sizeof(size_t) / sizeof(data[0]);
+	data_hash                   = 0;
+	size_t i                    = 0;
+	if (data.size() >= chunk_size)
+	{
+		for (; i <= (data.size() - chunk_size); i += chunk_size)
+		{
+			size_t it = 0;
+			std::memcpy(&it, &data[i], chunk_size);
+			glm::detail::hash_combine(data_hash, it);
+		}
+
+		// Handle elements smaller than sizeof(size_t) with zero padding
+		if (i < data.size())
+		{
+			size_t it = 0;
+			std::memcpy(&it, &data[i], data.size() - i);
+			glm::detail::hash_combine(data_hash, it);
+		}
+	}
```

Comment:
There's no need to `memcpy` all the data into a `size_t` before hashing. And you're missing the unlikely case of `data.size() < chunk_size`.
Maybe something like that:
```
	size_t offset = 0;
	for (; offset + chunk_size < data.size(); offset += chunk_size)
	{
		glm::detail::hash_combine(data_hash, *reinterpret_cast<size_t const *>(&data[offset]));
	}
	if (offset < data.size())
	{
		size_t it = 0;
		std::memcpy(&it, &data[offset], data.size() - offset);
		glm::detail::hash_combine(data_hash, it);
	}
```

Reply:
Perfect, updated, thank you!